### PR TITLE
METRON-2018 added docker tests for kafka plugin

### PR DESCRIPTION
**Pull request by ottobackwards**
This PR adds support for running tests when doing prepare-commit on a metron bro kafka plugin pr.

## Testing
run prepare commit on #32 and Y to run test suite.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [-] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [-] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```

- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

**Comment by ottobackwards** [comment]
@nickwallen can I get a quick review on this?
And ping @JonZeolla

File: dev-utilities/committer-utils/metron-committer-common [hunk 1]
```diff
@@ -434,6 +444,35 @@ function run_mvn_build_rpms {
...
+
+#
+#   runs the finish bro docker script to cleanup
+#
+function finish_bro_docker {
+    cd docker &> /dev/null || { echo "failed to change directory to docker"; exit 1; }
+    ./finish_end_to_end.sh
+
+    rc=$?; if [[ ${rc} != 0 ]]; then
+        echo "ERROR> FAILED run_end_to_end"
```
Comment (JonZeolla) [inline dev-utilities/committer-utils/metron-committer-common]: `s/run/finish/`

Comment (ottobackwards) [reply dev-utilities/committer-utils/metron-committer-common]: oops, thanks

**Comment by JonZeolla** [comment]
+1 by inspection

**Comment by nickwallen** [comment]
Ha!  Godfather.  I can certainly take it for a test spin, if you like.  Feel free to merge based on @JonZeolla 's +1 though.  I trust its good stuff. :)

**Comment by nickwallen** [comment]
I did get this error when spinning this up.  Do we really require Bash >4.0 for this?

```
$ /Users/nallen/tmp/metron-pr1348/dev-utilities/committer-utils/prepare-commit
/Users/nallen/.metron-prepare-commit
  ...using settings from /Users/nallen/.metron-prepare-commit
    [1] metron
    [2] metron-bro-plugin-kafka
  which repo? [1]: 2
  pull request: 31
  local working directory [/Users/nallen/tmp/metron-bro-plugin-kafka-pr31]:
  origin repo [https://github.com/apache/metron-bro-plugin-kafka]:
  base branch to merge into [master]:
Cloning into '/Users/nallen/tmp/metron-bro-plugin-kafka-pr31'...
remote: Enumerating objects: 29, done.
remote: Counting objects: 100% (29/29), done.
remote: Compressing objects: 100% (23/23), done.
remote: Total 342 (delta 10), reused 17 (delta 6), pack-reused 313
Receiving objects: 100% (342/342), 98.23 KiB | 831.00 KiB/s, done.
Resolving deltas: 100% (198/198), done.
From https://gitbox.apache.org/repos/asf/metron-bro-plugin-kafka
 * branch            master     -> FETCH_HEAD
 * [new branch]      master     -> upstream/master
Already on 'master'
Your branch is up to date with 'origin/master'.
Already up to date.
remote: Enumerating objects: 68, done.
remote: Counting objects: 100% (68/68), done.
remote: Total 87 (delta 68), reused 68 (delta 68), pack-reused 19
Unpacking objects: 100% (87/87), done.
From https://github.com/apache/metron-bro-plugin-kafka
 * [new ref]         refs/pull/31/head -> pr-31

  github contributor's username [JonZeolla]:
  github contributor's email [zeolla@gmail.com]:
  issue identifier in jira [METRON-2020]:
  issue description [Running run_end_to_end.sh with docker give warning if bash &lt; 4.0]:
  commit message [METRON-2020 Running run_end_to_end.sh with docker give warning if bash &lt; 4.0 (JonZeolla via nickwallen) closes apache/metron-bro-plugin-kafka#31]:

Auto-merging docker/run_end_to_end.sh
Squash commit -- not updating HEAD
Automatic merge went well; stopped before committing as requested
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean


  run test suite? [yN] y
  please verify that no bro docker containers are running before continuing,
  no bro docker containers are running, ready to proceed [yN] y
ERROR> bash >= 4.0 is required

USAGE
    --skip-docker-build             [OPTIONAL] Skip build of bro docker machine.
    --data-path                     [OPTIONAL] The pcap data path. Default: ./data
    --kafka-topic                   [OPTIONAL] The kafka topic to consume from. Default: bro
    -h/--help                       Usage information.

COMPATABILITY
     bash >= 4.0 is required.

ERROR> FAILED run_end_to_end
Running cleanup_containers with
CONTAINER_NAME = bro
NETWORK_NAME   = bro-network
===================================================
Running stop_container with
CONTAINER_NAME= bro
===================================================
Error response from daemon: No such container: bro
ERROR> FAILED finish_end_to_end
```
```
$ bash --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin18)
Copyright (C) 2007 Free Software Foundation, Inc.
```

**Comment by ottobackwards** [comment]
yes;  I think @JonZeolla can explain

**Comment by JonZeolla** [comment]
Yeah, that's the first version with associative arrays.

**Comment by JonZeolla** [comment]
FYI I used this to prepare and test the commit for METRON-2003 (apache/metron-bro-plugin-kafka#26) without issue.  The only thing that I didn't think of when I reviewed this earlier by inspection was that it took the time to rebuild my bro container from scratch. I don't mind this because when we use newer bro-pkg features in the tests or upgrade the version of bro that the tests use it would need to be rebuilt anyway, although I wouldn't push back on adding a y/n question on rebuilding the container.  Either as a follow-on or in this PR is fine with me.

**Comment by ottobackwards** [comment]
Can we do that as a follow on?

**Comment by nickwallen** [comment]
> @JonZeolla: Yeah, that's the first version with associative arrays.

So this was introduced by a previous PR?  

It is unfortunate to add another dependency hurdle like this.  Is there no way to avoid this?

**Comment by ottobackwards** [comment]
This was introduced during the development of the docker testing framework for the bro plugin.
Maybe we should check the bash version in the script and prompt that the bro tests cannot be run?

**Comment by nickwallen** [comment]
> Maybe we should check the bash version in the script and prompt that the bro tests cannot be run?

I think what we have clearly tells me that Bash <4.0.0 is the problem.  I'm good with the error messaging as it stands, just not that I have to upgrade Bash.

**Comment by nickwallen** [comment]
But if the requirement was introduced in a previous PR, then I can't complain about this one.  If that's the case I'd be a +0 on this one.

**Comment by ottobackwards** [comment]
Well, we are crossing the streams here @nickwallen, that previous pr was in the bro kafka plugin project, which is not the same as metron proper right

**Comment by nickwallen** [comment]
Ok, so it was introduced in a separate PR.  I'll go +0 on this.  Definitely useful functionality.  Thanks guys.

**Comment by ottobackwards** [comment]
@JonZeolla are you all set on this?

**Comment by JonZeolla** [comment]
Yup I'm still +1

**Pull request closed without merging**